app/src/main/java/com/example/series_collector/ui/adapters/CategoryAdapter.kt

```diff
@@ -0,0 +1,74 @@
+package com.example.series_collector.ui.adapters
+
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.*
+import com.example.series_collector.data.Category
+import com.example.series_collector.databinding.ListItemCategoryBinding
+import com.example.series_collector.utils.CATEGORY_FICTION
+import com.example.series_collector.utils.CATEGORY_TRAVEL
+
+
+class CategoryAdapter : ListAdapter<Category, RecyclerView.ViewHolder>(CategoryDiffCallback()) {
+
+    private val viewPool: RecyclerView.RecycledViewPool = RecyclerView.RecycledViewPool()
+    private lateinit var context: Context
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+        context = parent.context
+        return CategoryViewHolder(
+            ListItemCategoryBinding.inflate(
+                LayoutInflater.from(parent.context),
+                parent,
+                false
+            )
+        )
+    }
+
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        val category = getItem(position)
+        (holder as CategoryViewHolder).bind(
+            item = category,
+            context = context,
+            sharedPool = viewPool
+        )
+    }
+
+    class CategoryViewHolder(
+        private val binding: ListItemCategoryBinding,
+    ) : RecyclerView.ViewHolder(binding.root) {
+        private var snapHelper: SnapHelper = LinearSnapHelper()
+        private var adapter: SeriesAdapter = SeriesAdapter()
+
+        fun bind(item: Category, context: Context, sharedPool: RecyclerView.RecycledViewPool) {
+            binding.apply {
+                category = item
+                if (category?.categoryId == CATEGORY_TRAVEL || category?.categoryId == CATEGORY_FICTION) {
+                    rvSeries.layoutManager =
+                        GridLayoutManager(context, 2, RecyclerView.HORIZONTAL, false)
+                }
+                (rvSeries.layoutManager as LinearLayoutManager).recycleChildrenOnDetach = true
+                rvSeries.adapter = adapter
+                rvSeries.setRecycledViewPool(sharedPool)
+
+                snapHelper.attachToRecyclerView(rvSeries)
+
+                adapter.submitList(item.seriesList)
+
+                executePendingBindings()
```

Comment:
```suggestion
                executePendingBindings()
```
executePendingBindings 함수는 어떤 기능을 제공하며 내부 구현은 어떤식으로 되어있을까요?

Reply:
메인 스레드에서 다음 프레임이 오기전 예정되어있던 바인딩들을 즉시 실행시켜줍니다 :)